### DISCUSSION TITLE
Livewire 3 - wire:navigate / navigate between to components that have different layout

### DISCUSSION THREAD

**Discussion by sagadsalem** [livewire/livewire#6107] [7. Bugs (archived)] [answered]
### Livewire version

3.0@beta

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

When I navigate between two components that have different layout files with the livewire active on both layouts, 
and I use the new Navigate redirection, when I go from a page that already loads the scripts and styles in it layout to a page that will load the scripts and styles  I got a Warning
```
livewire.js?id=777821c0:8776 Detected multiple instances of Livewire running
livewire.js?id=777821c0:8778 Detected multiple instances of Alpine running
```

and when I do any functionality related to Livewire I got
```
TypeError: Alpine.navigate is not a function
```

my thoughts this is related to the conflict of Alpine and Livewire loaded two multiple times 
What i can do to prevent this?

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
Livewire now includes Alpinejs. Please remove any other Alpine references like CDN imports. This should solve the problem.

**Reply by sagadsalem** [reply]
@PhiloNL I don't have any Alpine installation on my page all that I have is the livewire scripts and styles on both layouts and also I turned off the inject option just to debug what is happening.

**Reply by sagadsalem** [reply]
@PhiloNL 
The Example for me is like this:
layouts/auth.blade.php with login.blade.php

when I sign in to my account I go to:
layouts/app.blade.php with dashboard.blade.php

until now All is Good but when I navigate between some pages and logout to return to the login page I get the warnings in the issue above.

**Comment by sagadsalem** [comment]
any help guys really have to make a decision here because time is not in my side.

**Reply by joshhanley** [reply]
@sagadsalem I don't recommend using V3 for anything that is important or time sensitive just yet, as it's still in beta so there will be bugs. This may be a bug with navigate and may not necessarily be fixed quickly. So I would be using V2 if it's time sensitive.

**Reply by sagadsalem** [reply]
@joshhanley thanks for the advice switching to v2 is the best option now but we will lose the spa mode in Livewire

hope this will be fixed soon .

**Comment by joshhanley** [comment]
@sagadsalem I've been looking into this issue but I can't seem to replicate it. Any chance you can provide both your layout files so I can run some tests? Thanks!

**Comment by WendellAdriel** [comment]
@joshhanley I think this and #6211 are the same so I'll post in both just in case.
I tried to recreate the issue in a demo app, but weirdly I can't recreate the behaviour on it.

Here's the repo if you or @sagadsalem want to use it as a demo app:
https://github.com/WendellAdriel/livewire-navigate-issue

**Reply by WendellAdriel** [reply]
For sure that I removed a lot of things, but this is kinda what I have in my project that's not working.

Two layouts, app and dashboard with almost the same content as the ones I have in the demo app, just some specific stuff and styling with TailwindCSS.

Tried to create it as simple as possible but also kinda trying to follow what I have in the project I'm getting this issue

**Reply by calebporzio** [reply]
To be clear, you're saying the linked repo DOESN"T reproduce the issue?

Also, can you try with the latest version of the main branch (composer update livewire/livewire:dev-main) to verify that your app isn't fixed? We're adding lots of wire:navigate fixes every day

**Reply by WendellAdriel** [reply]
Yeah, the linked repo is not reproducing the issue.
I didn't have time to work more on the demo repo to try to reproduce the issue yet.
I'll try with the main branch

**Reply by WendellAdriel** [reply]
@calebporzio I just updated to use dev-main and the issue is still there, but it's behaving in a diff way.

When I redirect to a page that shows those warnings in the console about multiple instances of Alpine and Livewire when I try to do any actions, it pops up an alert saying that the page expired, then I click ok and it does a full reload of the page.

**Reply by WendellAdriel** [reply]
I still didn't figure out why in some cases it's getting these multiple instances running tho.
I tried to look in the Livewire codebase but since I never worked on it before I still didn't figure out why that's happening.

**Comment by sagadsalem** [comment]
I tried the issue on a demo app but I think it's solved.

I Will try it in my live app and will give you feedback on the results.


@joshhanley @calebporzio @WendellAdriel

**Comment by sagadsalem** [comment]
All is GOOD!

I Navigate between two pages with different layouts files without any issues.


Thanks.

**Comment by WendellAdriel** [comment]
@sagadsalem is it ok for you?
I'm still finding one issue still that's related to this:
https://github.com/livewire/livewire/discussions/6211#discussioncomment-6689455

**Reply by sagadsalem** [reply]
It's work fine for my case. Without session regeneration

**Comment by WendellAdriel** [comment]
@calebporzio @joshhanley 
The issue with the session regeneration creating multiple instances can now be recreated with this repo: https://github.com/WendellAdriel/livewire-navigate-issue thanks to @thinkverse!!!

**Reply by sagadsalem** [reply]
@WendellAdriel I think so, because in my project I don't use regeneration session in the logout function in the last test

**Reply by thinkverse** [reply]
@sagadsalem seems there are multiple ways to get the same issue, either you're using `@livewireScripts` and `@livewireStyles` at the same time as Livewire is injecting its scripts and styles automatically, you've forgotten to remove, or are manually adding Alpine in your `app.js` file, and thirdly you are regenerating the session and then redirecting.

Seems those are the three ways we've found so far can cause the multiple instances issues.

**Comment by WendellAdriel** [comment]
@calebporzio @joshhanley do you recommend any strategy for avoiding this issue while recreating sessions?

**Comment by WendellAdriel** [comment]
@calebporzio @joshhanley Yesterday I was playing around to see if I could come to a solution to that
I was changing Livewire directly in my vendor files to see and one change that I did fix this issue

I tried to set up my local Livewire repo but as I tweeted: https://twitter.com/wendell_adriel/status/1691829044503793715
I was not able to make it work even following the guide for how to compile the JS locally
I also tried some other things but I can't make the `npm run build` script work for me locally, so I'm going to share here what I tested and you can see if this is a viable solution.

It was a really small and simple change but IDK the impacts of it, I tested in this repo that was having the issue: https://github.com/WendellAdriel/livewire-navigate-issue and after the change the issue stopped but IDK if this could cause any side effect and I was not able to test it properly because I was not able to make the Livewire repo to run for me 😓 

The change I did was in this file: https://github.com/livewire/livewire/blob/main/js/index.js

I removed these lines: https://github.com/livewire/livewire/blob/main/js/index.js#L25-L26

And update these lines: https://github.com/livewire/livewire/blob/main/js/index.js#L35-L36 to

```js
if (!window.Livewire) {
    window.Livewire = Livewire
}

if (!window.Alpine) {
    window.Alpine = Alpine
}
```

As I said, it fixed the issue, I didn't see any other issues in the pages but IDK if this could cause any side effects and I was not able to test it properly.

**Comment by calebporzio** [comment, marked as answer]
Thanks for the repository and help @WendellAdriel 

The issue has been fixed and described here: https://github.com/livewire/livewire/pull/6493

Will be available in the next release (today sometime)

**Reply by WendellAdriel** [reply]
Thank you so much for all the help with that @calebporzio!!! 💪

**Reply by sagadsalem** [reply]
@calebporzio thanks for the all work you guys do.
